nexus/src/nexus.rs

```diff
@@ -1736,6 +1745,27 @@ impl Nexus {
         Ok(())
     }
 
+    pub async fn vpc_fetch(
+        &self,
+        opctx: &OpContext,
+        organization_name: &Name,
+        project_name: &Name,
+        vpc_name: &Name,
+    ) -> LookupResult<db::model::Vpc> {
+        let authz_project = self
+            .db_datastore
+            .project_lookup_by_path(organization_name, project_name)
+            .await?;
+        Ok(self
+            .db_datastore
+            .vpc_fetch(opctx, &authz_project, vpc_name)
+            .await?
+            .1)
+    }
+
+    // TODO-security TODO-cleanup Remove this function.  Callers should use
```

Comment:
Similar question here: does it make sense to do this as part of this PR?

Reply:
In this case, this function is used by a lot of callers in a lot of code paths that aren't currently protected by authz.  Those will be updated in follow-up PRs, at which point it will be a lot easier to remove this.